### PR TITLE
xonsh

### PR DESCRIPTION
**Pull request by internaught**
new file:   Library/Formula/xonsh.rb

Adding xonsh to homebrew! http://xonsh.org

### CODE REVIEW

File: Library/Formula/xonsh.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  depends_on :python3
+
+  resource "ply" do
+    url "https://pypi.python.org/packages/source/p/ply/ply-3.6.tar.gz"
+    sha256 "61367b9eb2f4b819f69ea116750305270f1df8859992c9e356d6a851f25a4b47"
+  end
+
+  def install
+    version = Language::Python.major_minor_version "python3"
+    ENV["PYTHONPATH"] = lib/"python#{version}/site-packages"
```
Comment (tdsmith) [inline Library/Formula/xonsh.rb]: This line doesn't do anything and can be removed.

File: Library/Formula/xonsh.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python#{version}/site-packages"
+
```
Comment (tdsmith) [inline Library/Formula/xonsh.rb]: Please add:
`ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python#{version}/site-packages"`

File: Library/Formula/xonsh.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    resource("ply").stage do
+      system "python3", *Language::Python.setup_install_args(libexec)
```
Comment (tdsmith) [inline Library/Formula/xonsh.rb]: please install to `libexec/"vendor"`

**Comment by tdsmith** [comment]
Just FYI, our preferred commit message format for new formulas is like "xonsh 0.1.5 (new formula)"; if you make changes and squash your commits it'd be great if you could use that.

**Comment by internaught** [comment]
nvm

**Comment by internaught** [comment]
xonsh 0.1.5 (new formula)

**Pull request closed without merging**